pkg/checker/checker_test.go

```diff
@@ -245,7 +245,7 @@ metadata:
 	})
 	t.Run("Bad String", func(t *testing.T) {
 		nilChecker, err := NewCheckerFromString("i am a bad string@@: ", "")
-		assert.Nil(t, err)
+		assert.NotNil(t, err)
```

Comment:
why was this switched to NotNil?

Reply:
based on the following suggestion from 
https://github.com/rancher/hull/issues/35#issuecomment-1489176113
```
I also think part of this fix should be to remove the specific error check referenced above, with a comment that states that Parse is only expected to be used for a valid Kubernetes YAML manifest:
```
i removed the err check which was letting this test bypass and added a comment for the parser

Reply:
because now the parser will throw an error for invalid yaml string.